### ISSUE TITLE
Incorrect HTMLInfoWindow anchor for markers with custom image set as data: string 

### ISSUE THREAD

**Issue by marcinn** [mapsplugin/cordova-plugin-googlemaps#2549]
In `HtmlInfoWindow` there is a  `calculate` function, which contains a bug if marker's icon is set to `data:`.
```
    var icon = marker.get("icon");

    if (typeof icon === "object") {
      if (typeof icon.url === "string" && icon.url.indexOf("data:image/") === 0) 
```

In such case `icon` variable contains string `data:image/png....`, it is NOT an object, so the code inside the condition is not run. Icon size and anchor are **incorrect**, because `calculate` assumes this:

```    
var iconSize = {
      width: 62,
      height: 110
    };

    // If there is no specification with `anchor` property,
    // the values {x: 0.5, y: 1} are specified by native APIs.
    // For the case, anchor values are {x: 0} in JS.
    var anchor = {
      x: 15,
      y: 15
    };
```

and these values are used for further calculation. As a result infoWindow has wrong position, because it cannot read marker's icon anchor properly. In such case it should be taken from:
```
var anchor = marker.get("anchor")
```
because `icon` is a string and has no `anchor` property.

EDIT:
There is probably another bug for markers with Icon defined as an object, because `marker.setIconAnchor(x,y)` calls `marker.set("anchor", [x,y])`, so the `anchor` property of the `marker.icon` object is not changed. But the code inside `calculate()` fn is using just `marker.icon.anchor`.

**Comment by wf9a5m75** [comment]
Please use "as is", or send a pull request to fix the code. This plugin is fully open source ;)

**Comment by marcinn** [comment]
> Please use "as is", 

It is not working "as is".

> or send a pull request to fix the code

No. I see no possibility to make apps dependent on this plugin, because JS side is slow and plugin is too buggy. I decided to rewrite the app as a native Android/iOS.  

I've reported issues here, because I've spent many hours debugging them. I've shared what I've found and my thoughts about possible solution. That's all I can do.

Good luck!

Good luck!
